contracts/schemes/DxDaoSchemeConstraints.sol

```diff
@@ -0,0 +1,149 @@
+        }
+        contractsWhiteList = _contractsWhiteList;
+    }
+
+    /*
+     * @dev isAllowedToCall should be called upon a proposal execution.
+     *  - check that the total spending of tokens within a 'periodSize' does not exceed the periodLimit per token
+     *  - check that the total sending of eth within a 'periodSize' does not exceed the periodLimit
+     * @param _contractsToCall the contracts to be called
+     * @param _callsData - The abi encode data for the calls
+     * @param _values value(ETH) to transfer with the calls
+     * @param _avatar avatar
+     * @return bool value true-allowed false not allowed
+     */
+    function isAllowedToCall(
+        address[] calldata _contractsToCall,
+        bytes[] calldata _callsData,
+        uint256[] calldata _values,
+        Avatar
+    )
+    external
+    returns(bool)
+    {
+
+        uint256 observervationIndex = observationIndex();
+        uint256 totalPeriodSpendingInWei;
+        for (uint i = 0; i < _contractsToCall.length; i++) {
+        // constraint eth transfer
+            totalPeriodSpendingInWei = totalPeriodSpendingInWei.add(_values[i]);
+            bytes memory callData = _callsData[i];
+        // constraint approve calls
+            if (callData[0] == APPROVE_SIGNATURE[0] &&
+                callData[1] == APPROVE_SIGNATURE[1] &&
+                callData[2] == APPROVE_SIGNATURE[2] &&
+                callData[3] == APPROVE_SIGNATURE[3]) {
+                uint256 amount;
+                address contractToCall = _contractsToCall[i];
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    amount := mload(add(callData, 68))
+                }
+                periodSpendingToken[observervationIndex][contractToCall] =
+                periodSpendingToken[observervationIndex][contractToCall].add(amount);
+                require(
+                periodSpendingToken[observervationIndex][contractToCall] <= periodLimitToken[contractToCall],
+                "periodSpendingTokensExceeded");
+            }
+
+        }
+        periodSpendingWei[observervationIndex] =
+        periodSpendingWei[observervationIndex].add(totalPeriodSpendingInWei);
+        require(periodSpendingWei[observervationIndex] <= periodLimitWei, "periodSpendingWeiExceeded");
+        return true;
+    }
+
+    /*
+     * @dev isAllowedToPropose should be called upon a proposal submition.
+     * allow only whitelisted target contracts or 'approve' calls which the 'spender' is whitelisted
+     * @param _contractsToCall the contracts to be called
+     * @param _callsData - The abi encode data for the calls
+     * @param _values value(ETH) to transfer with the calls
+     * @param _avatar avatar
+     * @return bool value true-allowed false not allowed
+     */
+    function isAllowedToPropose(
+        address[] calldata _contractsToCall,
+        bytes[] calldata _callsData,
+        uint256[] calldata,
```

Comment:
uint256[] calldata > uint256[] calldata _values?

Reply:
yes, because the interface defined that function and there is no use for values at this function